gq/verify.go

```diff
@@ -73,8 +73,17 @@ func (sv *signerVerifier) Verify(proof []byte, identity []byte, message []byte)
 	return bytes.Equal(R, Rstar)
 }
 
+func (sv *signerVerifier) VerifyJWT(jwt []byte) bool {
+	signingPayload, signature, err := parseJWT(jwt)
+	if err != nil {
+		return false
+	}
+
+	return sv.Verify(signature, signingPayload, signingPayload)
+}
+
 func (sv *signerVerifier) decodeProof(s []byte) (R, S []byte, err error) {
-	bin, err := util.Base64Decode(s)
+	bin, err := util.Base64DecodeForJWT(s)
```

Comment:
This is super interesting! And also really confusing!  The reference code from the JWS spec you mentioned is equivalent to golang's `RawUrlEncoding`. I thought base64 encodings were reversible, but it sounds like they're not? Can you tell me more about which situations you were hitting where they weren't? Or maybe it has something to do with the conversions we're doing?

Reply:
Hmmm yes I think I've got confused at some point here, the code in this PR does not have the same issue I was hitting before. I had an issue when decoding, I would sometimes (rarely) decode 1 less byte than expected. I thought I fixed this by switching to padded base64 encoding but I think I must be misremembering because the length shouldn't be affected by this :thinking: 

Anyway, disregard this comment and let's merge this PR!